src/commands/cloud-run/flare.ts

```diff
@@ -228,25 +281,134 @@ export const getCloudRunServiceConfig = async (
  */
 export const maskConfig = (config: any) => {
   // We stringify and parse again to make a deep copy
-  const configCopy = JSON.parse(JSON.stringify(config))
+  const configCopy: IService = JSON.parse(JSON.stringify(config))
   const containers = configCopy.template?.containers
   if (!containers) {
     return configCopy
   }
 
-  for (const container of configCopy.template.containers) {
-    for (const envVar of container.env) {
-      if (!SKIP_MASKING_CLOUDRUN_ENV_VARS.has(envVar.name)) {
-        envVar.value = maskString(envVar.value)
+  for (const container of containers) {
+    const env = container.env ?? []
+    for (const envVar of env) {
+      if (!SKIP_MASKING_CLOUDRUN_ENV_VARS.has(envVar.name ?? '')) {
+        const val = envVar.value
+        if (val) {
+          envVar.value = maskString(val)
+        }
       }
     }
   }
 
   return configCopy
 }
 
+/**
+ * Gets recent logs
+ * @param projectId
+ * @param serviceId
+ * @param location
+ * @param isOnlyTextLogs whether or not to only get logs with a text payload
+ * @param severityFilter if included, adds the string to the filter
+ * @returns array of logs as CloudRunLog interfaces
+ */
+export const getLogs = async (
+  projectId: string,
+  serviceId: string,
+  location: string,
+  isOnlyTextLogs: boolean,
+  severityFilter?: string
+) => {
+  const logs: CloudRunLog[] = []
+  const logging = new Logging({projectId})
+
+  // Only get recent logs
+  const date = new Date()
+  date.setMinutes(date.getMinutes() - MAX_LOG_AGE_MINUTES)
+  const formattedDate = date.toISOString()
+
+  // Query options
+  let filter = `resource.labels.service_name="${serviceId}" AND resource.labels.location="${location}" AND timestamp>="${formattedDate}"`
+  filter += severityFilter ? ` AND ${severityFilter}` : ''
+  filter += isOnlyTextLogs ? ' AND textPayload:*' : ' AND (textPayload:* OR httpRequest:*)'
```

Comment:
We only care about logs that are an HTTP request or that have a text payload. Other logs are just audit logs which cause a lot of unnecessary spam.